### PR TITLE
🛂 Add `PassportIssuer` subgraph

### PR DESCRIPTION
**Pull request by acemasterjb**
## Summary

This PR pushes a very simple `PassportIssuer` subgraph that tracks:

```solidity
event Attest(address indexed _to, uint256 indexed _tokenId);
event Revoke(address indexed _to, uint256 indexed _tokenId);
event UpdateRequirements(uint256 claimRequiredBalance, uint256 revokeUnderBalance);
```

A formal deployment to Goerli and Mainnet is needed to use on this subgraph ~(out of scope of this PR)~.

### CODE REVIEW

**Comment by acemasterjb** [comment]
Addresses:

- https://github.com/nation3/snapshot-strategies/issues/5
- [Add handling of revoked passports to Snapshot strategy](https://app.dework.xyz/nation3/development-guild-43499?taskId=d111c11c-dfdb-48c1-b23a-3b9d6c57d026)

File: README.md [hunk 1]
```diff
@@ -1,2 +1,37 @@
...
+
+https://thegraph.com/hosted-service/subgraph/nation3/passportissuance
+
+---
+
+Refer to the following guide from [The Graph's docs](https://thegraph.com/docs/en/querying/querying-the-graph/) on how to execute a query remotely.
+
+## Contributing
+
+Refer to [this document from The Graph's docs](https://thegraph.com/docs/en/developing/creating-a-subgraph/) on how to set up your development environment for making changes to, and deploying new, subgraphs.
```
Comment (aahna-ashina) [inline README.md:37]:
```suggestion
Refer to [this document from The Graph's docs](https://thegraph.com/docs/en/developing/creating-a-subgraph/) on how to set up your development environment for making changes to, and deploying new, subgraphs.

npm install
npm run build
npm run deploy-local
```

**Review by aahna-ashina — DISMISSED** [review]
Should `passportissuance/build/` be added to `.gitignore`?

**Comment by acemasterjb** [comment]
> Should `passportissuance/build/` be added to `.gitignore`?

I left it there as a type of "lock file" aka "pinning". Since codegen/compilation can introduce breaking changes.

If this doesn't make sense, I could always try rebasing and adding `build` to `.gitignore` :)

**Comment by johnmark13** [comment]
This is cool, quick work. What I do not get yet is exactly how you are modelling the entities as they (for me) to closely mirror the events. Shouldn't there be an entity that is Passport, and maybe another for Citizens, that basically holds a passport collection (or multiple collections if it is better to forget normalisation and duplicate data)....

Similarly for Atest events, they neatly model a relationship between an attester and an atestee, to build a good graph, those should be concrete entities that the event than adds to an N..N relationship within. Maybe that is all to come later, but I think we should define the schema properly before handling the events.

File: passportissuance/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "passportissuance",
+  "license": "UNLICENSED",
```
Comment (aahna-ashina) [inline passportissuance/package.json]:
```suggestion
  "license": "GPL-3.0-or-later",
```

**Review by aahna-ashina — DISMISSED** [review]

**Review by johnmark13 — DISMISSED** [review]

**Comment by acemasterjb** [comment]
> how you are modelling the entities

@johnmark13 You raise some excellent points about the architecture of this subgraph and good practices for creating a GraphQL database.

However, since the point of this subgraph, and the currently open [snapshot strategy PR](https://github.com/nation3/snapshot-strategies/pull/6) was to address nation3/snapshot-strategies#5 I sought to use this subgraph simply to **provide logs data for [nation3/snapshot-strategies](https://github.com/nation3/snapshot-strategies)** so I suppose this is the main point of this PR and the tickets I'm addressing [here](https://github.com/nation3/subgraphs/pull/1#issuecomment-1875711203).

However, I agree with your statement here
> I think we should define the schema properly before handling the events

And will be happy to get on with modelling the entities at a later time after further discussions on the purpose of this subgraph :) .

**Comment by johnmark13** [comment]
Yes, that is why I changed my review and approved. It is a slight pain as we will require multiple PRs to Snapshot, but I think it makes most sense to get voting working ASAP, as you have said!

**Comment by acemasterjb** [comment]
@aahna-ashina

I just added some commits to address your suggestions.

Unit tests for `Revoke` and `Attest` can now be run using `npm run test`. This is explained in the README.

**Review by aahna-ashina — APPROVED** [review]

**Pull request merged**